### PR TITLE
Add SpannableString and SpannedString

### PR DESCRIPTION
**Pull request by hendraanggrian**
Conveniently get all spans with `SpannedString.kt`, and some operator methods in `SpannableString.kt`.

### CODE REVIEW

File: api/0.1.txt [hunk 1]
```diff
@@ -343,6 +343,14 @@ package androidx.os {
 
 package androidx.text {
 
+  public final class SpannableStringKt {
```
Comment (romainguy) [inline api/0.1.txt]: Don't modify this file

Comment (hendraanggrian) [reply api/0.1.txt]: I just reversed it. Are we supposed to change only the current.txt?

File: src/main/java/androidx/text/SpannableString.kt [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+import android.text.Spannable
+import android.text.SpannableString
+import android.text.SpannableStringBuilder
+import android.text.Spanned.SPAN_INCLUSIVE_EXCLUSIVE
+
+/**
+ * Returns a new [SpannableStringBuilder] from source,
+ * or the source itself if it is already an instance of [SpannableStringBuilder].
+ */
```
Comment (JakeWharton) [inline src/main/java/androidx/text/SpannableString.kt]: Documentation doesn't match the return type

File: src/main/java/androidx/text/SpannedString.kt [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+package androidx.text
+
+import android.text.Spanned
+import android.text.SpannedString
+
+/**
+ * Returns a new [SpannedString] from source,
+ * or the source itself if it is already an instance of [SpannedString].
+ */
```
Comment (JakeWharton) [inline src/main/java/androidx/text/SpannedString.kt]: Documentation doesn't match the return type

File: src/main/java/androidx/text/SpannedString.kt [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+inline fun CharSequence.toSpanned(): Spanned = SpannedString.valueOf(this)
+
+/** Returns `true` if [span] is found in this text. */
+inline operator fun Spanned.contains(span: Any): Boolean = getSpans<Any>().contains(span)
```
Comment (JakeWharton) [inline src/main/java/androidx/text/SpannedString.kt]: I'm a bit worried about the overhead of this method. Is this operation really something that we think people do?

Comment (hendraanggrian) [reply src/main/java/androidx/text/SpannedString.kt]: `Spanned.contains()` is probably not that common, I removed it in the latest commit. `CharSequence.toSpanned()` and `CharSequence.toSpannable()`, however, I imagine would be useful.

**Comment by JakeWharton** [comment]
Merged manually. Thanks!

**Comment by hendraanggrian** [comment]
Anytime!

**Pull request closed without merging**